### PR TITLE
send cocina JSON to dor_indexing_app

### LINKED ISSUE #3651: Call indexer with the cocina model when creating
https://github.com/sul-dlss/dor-services-app/blob/bab19e95747d6584bcaded3677d81cee9e750f0c/app/services/cocina_object_store.rb#L135

This will allow the indexer to skip calling-back to DSA to get the model.

See https://github.com/sul-dlss/dor_indexing_app/pull/769

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made? 🤔

fixes #3651 

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file system, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

- [x] unit tests
- [x] integration tests

ran integration tests against QA with sul-dlss/dor-services-app#3683 and sul-dlss/dor_indexing_app#803 deployed, all passed.

### CODE REVIEW

File: app/services/cocina_object_store.rb [hunk 1]
```diff
@@ -125,14 +125,16 @@ def create(cocina_request_object, assign_doi: false)
...
     add_tags_for_create(druid, cocina_object)
     # This creates version 1.0.0 (Initial Version)
     ObjectVersion.increment_version(druid)
 
+    created_at ||= Time.zone.now
+    updated_at ||= created_at
     # Fedora 3 has no unique constrains, so
     # index right away to reduce the likelyhood of duplicate sourceIds
-    SynchronousIndexer.reindex_remotely(druid)
+    SynchronousIndexer.reindex_remotely_from_cocina(druid, cocina_object: cocina_object, created_at: created_at, updated_at: updated_at)
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/services/cocina_object_store.rb:137]: I would agree...but won't it be easier to make that change now before this code is merged?  Currently nothing depends on that code, so it's easy to move around.

**Review by jcoyne — COMMENTED** [review]
I know the ticket doesn't ask for this, but would it also make sense to make this change in https://github.com/sul-dlss/dor-services-app/blob/2d741afde56ad244ad9b3dbd5a11312ead46be08/app/services/constituent_service.rb#L43 and then remove the old `reindex_remotely` code, since nothing else would be using this?  I think this would also be okay as a new issue,but I'd like your thoughts now while you're mind is on this problem.

**Comment by jmartin-sul** [comment]
converting to draft while i implement @jcoyne's suggestions (and rebase on `main`)

**Comment by jmartin-sul** [comment]
this is ready for another look, though marked it [HOLD] till https://github.com/sul-dlss/dor_indexing_app/pull/803 is merged

**Review by mjgiarlo — APPROVED** [review]
LGTM

**Comment by jmartin-sul** [comment]
merged since this was approved and the DIA PR it depends on was merged

**Pull request merged**